src/VisualStudio/Core/Def/Implementation/Preview/PreviewUpdater.cs

```diff
@@ -96,7 +125,9 @@ private void ApplyDocumentToBuffer(Document document, SpanChange spanSource, out
             }
 
             container = TextView.TextBuffer.AsTextContainer();
-            documentBackedByTextBuffer = document.WithText(container.CurrentText);
+            documentBackedByTextBuffer = isAdditionalDocument ?
+                document.WithAdditionalDocumentText(container.CurrentText) :
+                document.WithText(container.CurrentText);
```

Comment:
TextDocument is base type for all (source and non-source documents).
Document derives from TextDocument and represents all source documents.
AdditionalDocument: There is no strong type for additional document. It is basically a TextDocument that is not of type Document. Ideally, I would have preferred it to have a new type and TextDocument be abstract.